mobly/utils.py

```diff
@@ -257,6 +258,75 @@ def rand_ascii_str(length):
 
 
 # Thead/Process related functions.
+def collect_process_tree(starting_pid):
+  """Collects PID list of the descendant processes from the given PID.
+
+  This function only available on Unix like system.
+
+  Args:
+    starting_pid: The PID to start recursively traverse.
+
+  Returns:
+    A list of pid of the descendant processes.
+  """
+  ret = []
+  stack = [starting_pid]
+
+  while stack:
+    pid = stack.pop()
+    try:
+      ps_results = subprocess.check_output([
+          'ps',
```

Comment:
I see a couple of raw commands added.
do these commands work on windows?
and are they the same between mac and linux?

Reply:
As the doc string, this function only available on Unix like system.
The command works on Unix like system as you can see all tests pass on Windows, Mac and Linux.